cid-decode/src/codec.rs

```diff
@@ -0,0 +1,76 @@
+// Integrated from rust-cid v0.5.1 since they drop support
+// for codec in version 0.6.0
+
+use std::convert::TryFrom;
+
+use cid::{Error, Result};
+
+macro_rules! build_codec_enum {
+    {$( #[$attr:meta] $code:expr => $codec:ident, )*} => {
+        /// List of types currently supported in the multicodec spec.
+        #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug, Hash)]
+        pub enum Codec {
+            $( #[$attr] $codec, )*
+        }
+
+        impl TryFrom<u64> for Codec {
+            type Error = Error;
+
+            /// Convert a number to the matching codec, or `Error` if unknown codec is matching.
+            fn try_from(raw: u64) -> Result<Codec> {
+                match raw {
+                    $( $code => Ok(Self::$codec), )*
+                    _ => Err(Error::UnknownCodec),
+                }
+            }
+        }
+
+        impl From<Codec> for u64 {
+            /// Convert to the matching integer code
+            fn from(codec: Codec) -> u64 {
+                match codec {
+                    $( Codec::$codec => $code, )*
+                }
+            }
+        }
+    }
+}
+
+build_codec_enum! {
```

Comment:
This is a good start, but we'll need [the full table](https://github.com/multiformats/multicodec/blob/master/table.csv).
We could download and throw that in with the sources, include it with `include_str` or something like that, and parse it during compile time, maybe? Maybe in a `build.rs`? Lots of things to explore!